api/filters/basic_authentication.go

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2018 The Service Manager Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package filters
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/Peripli/service-manager/pkg/security/filters"
+
+	"github.com/Peripli/service-manager/pkg/types"
+
+	"github.com/Peripli/service-manager/pkg/query"
+
+	httpsec "github.com/Peripli/service-manager/pkg/security/http"
+
+	"github.com/Peripli/service-manager/pkg/web"
+	"github.com/Peripli/service-manager/storage"
+)
+
+const BasicAuthnFilterName string = "BearerAuthnFilter"
+
+func NewBasicAuthnFilter(repository storage.Repository) *filters.AuthenticationFilter {
+	return filters.NewAuthenticationFilter(&basicAuthenticator{
+		Repository: repository,
+	}, BasicAuthnFilterName, basicAuthnMatchers())
+}
+
+type basicAuthnData struct {
+	data json.RawMessage
+}
+
+func (bad *basicAuthnData) Data(v interface{}) error {
+	return json.Unmarshal([]byte(bad.data), v)
+}
+
+// basicAuthenticator for basic security
+type basicAuthenticator struct {
+	Repository storage.Repository
+}
+
+// Authenticate authenticates by using the provided Basic credentials
+func (a *basicAuthenticator) Authenticate(request *http.Request) (*web.UserContext, httpsec.Decision, error) {
+	username, password, ok := request.BasicAuth()
+	if !ok {
+		return nil, httpsec.Abstain, nil
+	}
+
+	ctx := request.Context()
+	byUsername := query.ByField(query.EqualsOperator, "username", username)
```

Comment:
the sbproxy has in-memory implementation of credentials storage that would require a dummy implementation of the whole repository

Reply:
not necessarily - i refactored the authenticatiom so that the pkg basic auth filter can be used with a custom inmemory authenticator

Reply:
proxy will call this method with custom inmemory authenticator  
```
func NewAuthenticationFilter(authenticator http.Authenticator, name string, matchers []web.FilterMatcher) *AuthenticationFilter {
	return &AuthenticationFilter{
		Authentication: &middlewares.Authentication{
			Authenticator: authenticator,
		},
		matchers: matchers,
		name:     name,
	}
}
```

Reply:
https://github.com/Peripli/service-broker-proxy/pull/77